docs/cookbook/creating-custom-schema-fields.md

```diff
@@ -0,0 +1,999 @@
+
+```javascript
+export default {
+  name: 'InputColorGradient',
+  components: {
+    AposInputWrapper,
+    AposSchema
+  },
+  mixins: [AposInputMixin, AposInputChoicesMixin],
+  ```
+
+The main script starts with the `name` property, which is set by convention to the same value as the name of the file. This is followed by the `components` property, with an object of the components that we just loaded. If your component uses the button component, you would add `AposButton` to this object. Similarly, the `mixin` property takes an array of the mixins we just imported.
+
+```javascript
+props: {
+  generation: {
+    type: Number,
+    required: false,
+    default() {
+      return null;
+    }
+  }
+  ```
+
+Next, we declare the `generation` prop that Apostrophe uses for triggering the re-render of the component, such as when the user changes page-type. 
+
+```javascript
+data() {
+  const next = this.getNext();
+  const gradeSchema = apos.modules['grade'].gradeSchema;
+  return {
+    next,
+    gradeSchemaInput: {
+      data: next
+    },
+    gradeSchema: gradeSchema
+  };
+},
+```
+
+In this code, we are defining the initial state of our `AposSchema` Vue component using the data option. The data option is a function that returns an object with properties that represent the component's initial state.
+
+First, we call a self-defined method called `getNext()` to retrieve the current values for our schema fields and assign them to the `next` constant.
+
+Next, we access the schema we used in our `grade` piece through the `apos.modules` object. Remember that we extended the `getBrowserData()` method to add that schema to the `req` as `gradeSchema`.
+
+Finally, we return an object with three properties:
+* `next`: the value of next
+* `gradeSchemaInput`: an object with a property called data that contains the value of `next`. This is used to set the `v-model` of the `AposSchema` component
+* `gradeSchema`: the schema of our custom field as an array. This is used to set the `:schema` of the `AposSchema` component
+  
+The next block of code sets up watchers to monitor and trigger an update if the input fields are changed.
+
+```javascript
+watch: {
+  generation() {
+    this.next = this.getNext();
+    this.gradeSchemaInput = {
+      data: this.next.value
+    };
+  },
+  gradeSchemaInput: {
+    deep: true,
+    handler() {
+      if (!this.gradeSchemaInput.hasErrors) {
+        this.next = this.gradeSchemaInput.data;
+      };
+    }
+  }
+},
+```
+
+The first watcher is `generation()`. This method will fire if the `generation` prop is either updated within the component or by Apostrophe. This causes the stored values for our schema fields to be updated. Note that you can utilize this watcher to trigger an update of the component. But, you should avoid mutating the passed prop by setting ``:generation=$"`${generation}:${innerGeneration}`"``, and then triggering the update by incrementing the `innerGeneration` value.
```

Comment:
How about: "this causes the component and all of its sub-fields to re-render, which is needed if a parent component needs to change the schema itself, e.g. when a field type is used in a page and the page type changes."

Reply:
Updated.